src/storage/core.rs

```diff
@@ -267,6 +267,26 @@ where
         self.write_with_optional_meta(key, value, Some(meta)).await
     }
 
+    /// Free all resources that may be freed without work interruption
+    /// NOTICE! This function frees part of the resources in separate thread,
+    /// so actual resources may be freed later
+    pub async fn free_freeable_resources(&self) {
+        self.observer.try_dump_old_blob_indexes().await;
+        self.close_active_blob_in_background().await;
+    }
+
+    /// Get size in bytes of inactive indexes
+    pub async fn inactive_index_memory(&self) -> usize {
+        let safe = self.inner.safe.read().await;
+        let blobs = safe.blobs.read().await;
+        blobs.iter().fold(0, |s, n| s + n.index_memory())
+    }
+
+    /// Get size in bytes of all freeable resources
+    pub async fn freeable_resources_memory(&self) -> usize {
```

Comment:
`index_memory` is a better name. The function with that name was previously used by `index_memory` metric, so it is good idea to keep the name, but take index memory from inactive blobs into account